crates/teloxide/src/dispatching/dialogue/get_chat_id.rs

```diff
@@ -23,3 +26,89 @@ impl GetChatId for Update {
         self.chat().map(|chat| chat.id)
     }
 }
+
+impl GetChatId for Recipient {
+    fn chat_id(&self) -> Option<ChatId> {
+        match self {
+            Recipient::Id(chat_id) => Some(*chat_id),
+            Recipient::ChannelUsername(_) => None,
+        }
+    }
+}
+
+impl GetChatId for BotCommandScope {
+    fn chat_id(&self) -> Option<ChatId> {
+        match self {
+            BotCommandScope::Default
+            | BotCommandScope::AllPrivateChats
+            | BotCommandScope::AllGroupChats
+            | BotCommandScope::AllChatAdministrators => None,
+            BotCommandScope::Chat { chat_id: recipient }
+            | BotCommandScope::ChatAdministrators { chat_id: recipient }
+            | BotCommandScope::ChatMember { chat_id: recipient, .. } => recipient.chat_id(),
+        }
+    }
+}
+
+impl GetChatId for Chat {
```

Comment:
Actually, `Chat` has it's id without `Option`, so it's better to use the field directly

Reply:
True, but same as above, can't we have it implemented anyways? Since it allow users to use `impl GetChatId` in parameters (`impl Trait` in argument position). I see this a small benefit